src/features/query-bar/flowTypes.js

```diff
@@ -3,7 +3,7 @@ import { AND, OR } from './constants';
 
 export type SelectOptionType = {
     displayText: string,
-    fieldId?: string,
+    fieldId: string,
     type?: string,
```

Comment:
fieldId should always be present given that a Column has been selected

Reply:
If a Template has not been selected, then the FilterButton would be disabled and thus not have to render any options